src/sabre_swap/mod.rs

```diff
@@ -154,18 +160,85 @@ pub fn build_swap_map(
     heuristic: &Heuristic,
     seed: u64,
     layout: &mut NLayout,
-) -> PyResult<(SwapMap, PyObject)> {
-    let mut gate_order: Vec<usize> = Vec::with_capacity(dag.dag.node_count());
+    num_trials: usize,
+) -> (SwapMap, PyObject) {
     let run_in_parallel = getenv_use_multiple_threads();
-    let mut out_map: HashMap<usize, Vec<[usize; 2]>> = HashMap::new();
-    let mut front_layer: Vec<NodeIndex> = dag.first_layer.clone();
+    let dist = distance_matrix.as_array();
+    let coupling_graph: DiGraph<(), ()> = cmap_from_neighor_table(neighbor_table);
+    let outer_rng = Pcg64Mcg::seed_from_u64(seed);
+    let seed_vec: Vec<u64> = outer_rng
+        .sample_iter(&rand::distributions::Standard)
+        .take(num_trials)
+        .collect();
+    let result = if run_in_parallel {
+        // Collect into an intermediate Vec for deterministic ordering.
+        // If min_by_key() were used directly on the parallel iterator
+        // (which would be more efficient) then for trials with equal
+        // minimum swap counts the output selected would potentially
+        // vary based on the execution time between the parallel trials.
+        let trial_results: Vec<TrialResult> = seed_vec
+            .into_par_iter()
+            .map(|seed_trial| {
+                swap_map_trial(
+                    num_qubits,
+                    dag,
+                    neighbor_table,
+                    &dist,
+                    &coupling_graph,
+                    heuristic,
+                    seed_trial,
+                    layout.clone(),
+                )
+            })
+            .collect();
+        trial_results
```

Comment:
Another idea might be to map the seed vec with index, and then compare by key as the tuple of the index and the trial result.

Reply:
Ooh, I like this idea. I think in practice it won't make a huge difference (except at large trial sizes) but adding the intermediate `Vec` is something I did begrudgingly so being able to remove it is a win in my book

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8572/commits/24903abe4bf348a8de6ad639f43ccfcdbfde6108